### PR TITLE
Replace CUDA 11.1 Linux CI with CUDA 11.2

### PR DESCRIPTION
**Pull request by janeyx99**
Adding 11.2 to CI with BUILD_SPLIT_CUDA enabled.

Disabled the following tests as they were failing in test_optim.py:
test_adadelta
test_adam
test_adamw
test_multi_tensor_optimizers
test_rmsprop

(Issue tracking that is here: https://github.com/pytorch/pytorch/issues/51992)

### CODE REVIEW

File: test/test_optim.py [hunk 1]
```diff
@@ -379,7 +380,7 @@ def test_multi_tensor_optimizers(self):
             for p1, p2 in zip(res[0], res[1]):
                 self.assertEqual(p1, p2)
 
-
+    @unittest.skipIf(True, "test does not pass for CUDA 11.2")
```
Comment (walterddr) [inline test/test_optim.py:383]: nit: Link github issue here with comments to #51598?

**Review by walterddr — APPROVED** [review]
lgtm. windows CI failures seems irrelevant?

File: test/test_optim.py (hunk 1 above)
Comment (janeyx99) [reply test/test_optim.py:383]: I added the link to the issue in the description of this PR. Interestingly, these same tests do now fail for windows

**Comment by samestep** [comment]
Blocked by #52054.

**Pull request closed without merging**